packages/core/src/session.ts

```diff
@@ -332,6 +338,11 @@ class Session {
     return this._lastBookmarks.values()
   }
 
+  private _bookmarks (): Bookmarks {
+    const bookmarks = this._bookmarkManager?.getAllBookmarks() ?? []
+    return new Bookmarks([...bookmarks, ...this._lastBookmarks])
```

Comment:
If I read this all correctly, this is not exactly how the ADR describes it:
Once the session received bookmarks from the server, it should only rely on the bookmarks of the BMM, but no longer on its own bookmarks. This will make a difference, if there is another session (`s2`) running for the same db that fetches `s1._lastBookmarks` via the BMM and completes a TX replacing `s1._lastBookmarks` with `s2._lastBookmarks` in the BMM. This code will still send `s1._lastBookmarks` over the wire. It's not incorrect as the server will ignore the outdated bookmarks but it's wasteful still.

Reply:
Sure, the ADR was changed after the code is already implemented.